Plugins/NativeCamera/Source/Android/NativeCameraImpl.cpp

```diff
@@ -54,88 +65,103 @@ namespace Babylon::Plugins
 
     std::string Camera::Impl::GetCameraId(bool frontCamera)
     {
-        ACameraIdList *cameraIds = nullptr;
-        ACameraManager_getCameraIdList(m_cameraManager, &cameraIds);
+        API24::ACameraIdList *cameraIds = nullptr;
+        GET_CAMERA_FUNCTION(ACameraManager_getCameraIdList)(m_cameraManager, &cameraIds);
 
         std::string cameraId{};
 
         for (int i = 0; i < cameraIds->numCameras; ++i)
         {
             const char *id = cameraIds->cameraIds[i];
 
-            ACameraMetadata *metadataObj;
-            ACameraManager_getCameraCharacteristics(m_cameraManager, id, &metadataObj);
+            API24::ACameraMetadata *metadataObj;
+            GET_CAMERA_FUNCTION(ACameraManager_getCameraCharacteristics)(m_cameraManager, id, &metadataObj);
 
-            ACameraMetadata_const_entry lensInfo = {};
-            ACameraMetadata_getConstEntry(metadataObj, ACAMERA_LENS_FACING, &lensInfo);
+            API24::ACameraMetadata_const_entry lensInfo = {};
+            GET_CAMERA_FUNCTION(ACameraMetadata_getConstEntry)(metadataObj, API24::ACAMERA_LENS_FACING, &lensInfo);
 
-            auto facing = static_cast<acamera_metadata_enum_android_lens_facing_t>(lensInfo.data.u8[0]);
+            auto facing = static_cast<API24::acamera_metadata_enum_android_lens_facing_t>(lensInfo.data.u8[0]);
 
             // Found a corresponding facing camera?
-            if (facing == (frontCamera ? ACAMERA_LENS_FACING_FRONT : ACAMERA_LENS_FACING_BACK))
+            if (facing == (frontCamera ? API24::ACAMERA_LENS_FACING_FRONT : API24::ACAMERA_LENS_FACING_BACK))
             {
                 cameraId = id;
                 break;
             }
         }
 
-        ACameraManager_deleteCameraIdList(cameraIds);
+        GET_CAMERA_FUNCTION(ACameraManager_deleteCameraIdList)(cameraIds);
         return cameraId;
     }
 
+    void* Camera::Impl::GetCameraDynamicFunction(const char* functionName)
```

Comment:
I'd be more inclined to just make this a global function and put it up by `libCamera2NDK`. I think storing the map globally is fine as well. Any specific reason to have it scoped to an instance of NativeCameraImpl?

Reply:
In the latest iteration I've moved all of the dynamic invocation logic into a CameraWrappers.h file to better consolidate it.